packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/AmplifyApiPlugin.kt

```diff
@@ -40,13 +42,48 @@ class AmplifyApiPlugin : FlutterPlugin, MethodCallHandler {
         LOG.info("Initiated API plugin")
     }
 
-    override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
-        when (call.method) {
-            "query" ->
-                FlutterGraphQLApi.query(result, call.arguments as Map<String, Any>)
-            "mutate" ->
-                FlutterGraphQLApi.mutate(result, call.arguments as Map<String, Any>)
-            else -> result.notImplemented()
+    override fun onMethodCall(call: MethodCall, result: Result) {
+        var methodName = call.method
+
+        if(methodName == "cancel"){
+            onCancel(result, (call.arguments as String))
+            return
+        }
+
+        try {
+            var arguments : Map<String, Any> = call.arguments as Map<String,Any>
+
+            when (call.method) {
+                "get" -> FlutterRestApi.onGet(result, arguments)
+                "post" -> FlutterRestApi.onPost(result, arguments)
+                "put" -> FlutterRestApi.onPut(result, arguments)
+                "delete" -> FlutterRestApi.onDelete(result, arguments)
+                "head" -> FlutterRestApi.onHead(result, arguments)
+                "patch" -> FlutterRestApi.onPatch(result, arguments)
+                "query" -> FlutterGraphQLApi.query(result, arguments)
+                "mutate" -> FlutterGraphQLApi.mutate(result, arguments)
+                else -> result.notImplemented()
+            }
+        } catch (e: ClassCastException) {
+            FlutterApiError.postFlutterError(
+                    result,
+                    FlutterApiErrorMessage.ERROR_CASTING_INPUT_IN_PLATFORM_CODE.toString(),
+                    e
+            )
+        }
+    }
+
+    fun onCancel(
+            flutterResult: Result,
+            cancelToken: String) {
+        if(OperationsManager.containsOperation(cancelToken)) {
+            OperationsManager.cancelOperation(cancelToken)
+            flutterResult.success("Operation Canceled")
+        } else {
+            flutterResult.error(
+                    "AmplifyAPI-CancelError",
```

Comment:
I know error handling is being worked on, but in all plugins, for now the code has been `AmplifyException` 
Are you using this cancel specific code on the Flutter side?

Reply:
No we don't it's just internal to our code.  The exception is just caught on the Flutter Dart side.

Reply:
As discussed offline we can create the Exception for that case and pass it in